tripy/tripy/backend/compiler_api.py

```diff
@@ -33,6 +33,110 @@
 from tripy.utils import json as json_utils
 
 
+@export.public_api(document_under="compiler")
+def synchronize():
+    """
+    Synchronizes the default stream.
+
+    .. code-block:: python
+        :linenos:
+        :caption: Use streams in execution
+
+        linear = tp.Linear(2, 3)
+        compiler = tp.Compiler(linear)
+
+        compiled_linear = compiler.compile(tp.InputInfo((2, 2), dtype=tp.float32))
+
+        a = tp.ones((2, 2), dtype=tp.float32)
+
+        out = compiled_linear(a)
+        tp.synchronize()
+        assert cp.array_equal(cp.from_dlpack(out), cp.from_dlpack(linear(a)))
+    """
+    Stream.get_default_stream().synchronize()
+
+
+@export.public_api(document_under="compiler")
+class Stream:
+    """
+    Represents a CUDA stream that can be used to manage concurrent operations.
+
+    This class is a wrapper around the underlying stream object, allowing management of CUDA streams.
+
+    Args:
+        priority (int): Assign priority for the new stream. Lower number signifies higher priority.
+
+    .. code-block:: python
+        :linenos:
+        :caption: Create new streams
+
+        streamA = tp.Stream()
+        streamB = tp.Stream()
+        defaultStream = tp.Stream.get_default_stream()
+        assert streamA != streamB
+        assert defaultStream != streamB
+
+    .. code-block:: python
+        :linenos:
+        :caption: Use streams in execution
+
+        linear = tp.Linear(2, 3)
+        compiler = tp.Compiler(linear)
+
+        compiled_linear = compiler.compile(tp.InputInfo((2, 2), dtype=tp.float32))
+
+        a = tp.ones((2, 2), dtype=tp.float32)
+
+        compiled_linear.stream = tp.Stream()
+        out = compiled_linear(a)
+        assert cp.array_equal(cp.from_dlpack(out), cp.from_dlpack(linear(a)))
+    """
+
+    _default_stream = None  # class variable
```

Comment:
Why isn't this just a global?

Reply:
Isn't it much cleaner for it to be part of the Stream class? What's the advantage of making it global variable?

Reply:
Global is easier to reason about IMO. A default stream is just a `Stream` that's created and available by default in Tripy, i.e. a global of type `Stream`. Making `Stream` contain a `Stream` as a class variable doesn't really align with that mental model. 

Besides, since you have:
```py
# Create default stream.
Stream.get_default_stream()
```
in the global scope, the default stream effectively is a global anyway. 

Even in PyTorch, the [default stream](https://pytorch.org/docs/stable/generated/torch.cuda.default_stream.html) is separated from the `Stream` class.